plugin/rename.py

```diff
@@ -226,3 +340,17 @@ def initial_text(self) -> str:
 
     def validate(self, name: str) -> bool:
         return len(name) > 0
+
+
+class LspHideRenameButtonsCommand(sublime_plugin.WindowCommand):
+
+    def run(self) -> None:
+        wm = windows.lookup(self.window)
+        if not wm:
+            return
+        pm = wm.panel_manager
+        if not pm:
+            return
+        buttons = pm.rename_panel_buttons
+        if buttons:
+            buttons.update([])
```

Comment:
Do we need this functionality to hide the buttons?

What I find weird is that pressing Discard hides the buttons and the panel but you can still reopen the panel and then you won't see the buttons but you will still see the preview results.

Reply:
The reason is that if the buttons wouldn't disappear, you could manually close the panel (<kbd>Esc</kbd>), make some more edits to the files, open the panel again 10 minutes later and if you click on the "Apply" button you end up with mostlikely broken changes applied to your files.

This is related to my comment https://github.com/sublimelsp/LSP/pull/2428#issuecomment-1975269097 and the same could happen currently if you just ignore the rename dialog, but click on "Replace" at some later time. As mentioned, I think it would be better to reject (all) edits if any of the file changes, but most like that needs more refactoring. I already thought about it a little bit, and currently we use failure handling "abort" strategy for workspace edits:
https://github.com/sublimelsp/LSP/blob/57c01e7f33a548a1da515d0805e7ce944bdc28a7/plugin/core/sessions.py#L462

I think it would be better to switch to `"textOnlyTransactional"` (or `"transactional"`), if we can implement it. See https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#failureHandlingKind